lib/rules/selector-pseudo-class-allowed-list/index.mjs

```diff
@@ -26,55 +31,62 @@ const rule = (primary) => {
 			possible: [isString, isRegExp],
 		});
 
-		if (!validOptions) {
-			return;
-		}
-
-		root.walkRules((ruleNode) => {
-			if (!isStandardSyntaxRule(ruleNode)) {
-				return;
-			}
-
-			if (!ruleNode.selector.includes(':')) {
-				return;
-			}
+		if (!validOptions) return;
 
-			parseSelector(getRuleSelector(ruleNode), result, ruleNode)?.walkPseudos((pseudoNode) => {
-				const value = pseudoNode.value;
-				const index = pseudoNode.sourceIndex;
-				const endIndex = index + pseudoNode.value.length;
+		/**
+		 * @param {string} selector
+		 * @param {AtRule | Rule} node
+		 */
+		function check(selector, node) {
+			const offset = node.type === 'atrule' ? atRuleParamIndex(node) : 0;
```

Comment:
In other `check` functions we pass the function to calculate the index as an argument.

Can we do the same here?

Reply:
I am not sure why it would be desirable.
I mean if I understand correctly the 3rd argument would be a function which means to get `0`, I would have to pass `() => 0` correct?
i.e. `const offset = fn(node);`

Or did you mean pass _the result_ of the function?

Reply:
No, pass the function itself.
Like we do in:

https://github.com/stylelint/stylelint/blob/2b603bbc55dc8ece7ac7022e907d415604b1fa8c/lib/rules/string-no-newline/index.mjs#L31-L36

Reply:
So you want us to pass `() => 0` for the `rule` case?

Reply:
Yeah, that is indeed annoying.
Fine to keep as is, we also have rules using this exact code.